R/parallelization.R

```diff
@@ -1,9 +1,64 @@
 #' Parallelization in purrr
 #'
 #' @description
-#' purrr's map functions have a `.parallel` argument to parallelize a map using
-#' the \CRANpkg{mirai} package. This allows you to run computations in parallel
-#' using more cores on your machine, or distributed over the network.
+#' `r lifecycle::badge("experimental")`
+#'
+#' All map functions allow parallelized operation using \CRANpkg{mirai}. To take
+#' advantage of this, wrap a function that is passed to the `.f` argument of
+#' [map()] or any of its variants with [in_parallel()]. This declares that the
+#' computation should proceed in parallel utilizing multiple cores on your local
+#' machine, or distributed over the network.
+#'
+#' @param .f A fresh formula or function. "Fresh" here means that they should be
+#'   declared in the call to [in_parallel()].
+#' @param ... Arguments to declare in the environment of the function.
+#'
+#' @section How to Use:
+#'
+#' Wrapping a function that is passed to the `.f` argument in [map()] or any of
+#' its variants with [in_parallel()] declares that the map should be performed
+#' in parallel.
+#'
```

Comment:
Suggestion: Structure the doc along these lines:


@description
`in_parallel()` is a purrr adverb that plays two roles:

- It is a _signal_ to purrr verbs like [map()] to go ahead and perform computations in parallel.
- It helps you create _self-contained_ functions that are isolated from your workspace. This is important because the function is packaged up (in technical terms serialized) to be sent over to worker processes. Isolation is critical for performance because it prevents sending very large objects unexpectedly.

@section Creating self-contained functions: